### ISSUE TITLE
System info probe returns confusing data

### ISSUE THREAD

**Issue by jan-cerny** [OpenSCAP/openscap#998]
System info probe in OpenSCAP 1.2.16 returns some values in OVAL results `system_info` element that I think could be improved to better inform the consumers of OVAL results.

In normal mode, when we scan local host, we get 
```
<system_info>
  <os_name>Linux</os_name>
  <os_version>#1 SMP Wed Feb 28 17:53:39 UTC 2018</os_version>
  <architecture>x86_64</architecture>
  <primary_host_name>thinkpad</primary_host_name>
  <interfaces>
     ...
```
I found weird that we return this date as OS version, but it is consistent with output of `uname --kernel-version`. Maybe we could add some information about Linux distribution used.

In offline mode, when scanning a RHEL7.4 virtual domain using oscap-vm:
```
<system_info>
  <os_name>Red</os_name>
  <os_version>Hat Enterprise Linux Server (3.10.0-693.el7.x86_64) 7.4 (Maipo)</os_version>
  <architecture>x86_64</architecture>
  <primary_host_name>localhost.localdomain</primary_host_name>
  <interfaces/>
</system_info>
```
Notice not only that the os name and version is broken, but it is not consistent with output in online mode (see above). Moreover, network interfaces information are missing. I also don't like that the probe is parsing /boot/grub/ files.

In offline scanning of Docker containers, it returns unknown.
```
<system_info>
  <os_name>Unknown</os_name>
  <os_version>Unknown</os_version>
  <architecture>Unknown</architecture>
  <primary_host_name>Unknown</primary_host_name>
  <interfaces/>
</system_info>
```
Some data aren't possible to get from an container, but if we provide data about Linux kernel on the host, we could maybe return it the same when scanning a container, because containers share the same kernel with the host.

OVAL specification in  Core System Characteristics schema is quite unspecific. 

> The required os_name elements describes the operating system of the machine the data was collected on.

> The required os_version elements describe the operating system version of the machine the data was collected on.

**Comment by evgenyz** [comment]
@jan-cerny 

Any idea why we are using --kenel-version to fill *os*_version field? And why it is not, at least, --kernel-release?

Wouldn't it be more sane to use information from '/etc/os-release'?

**Comment by evgenyz** [comment]
@jan-cerny 

Also, is there any particular reason to get os name/version information from `/boot/grub2/grubenv` in the offline mode? Why not `/etc/os-release' again?

**Comment by jan-cerny** [comment]
I have no idea.

I guess that grubenv is used because it's the place where it's possible to obtain information about architecture to populate the architecture filed.

Looks like it's introduced by https://github.com/OpenSCAP/openscap/pull/559 , which says it's a temporary fix, but there is no explanation.

**Comment by evgenyz** [comment]
OK. So, I dove deep into this probe, the history of fix #559 and current state of things. I have couple of ideas on how to clean this up.

First of all, `os_name` and `os_version` fields are filled with somewhat incorrect information in both modes.

In online mode they are filled with kernel details (which is not a whole OS by any definition). The **name** is generally filled with 'Linux', which is just a part of a part of the OS name. It could be, for example, 'Fedora (GNU/Linux)'. The **version** is even worse, as it does contain only kernel build timestamp (like 'SMP Mon Jul 15 15:42:34 UTC 2019'), which is not helpful for virtually anything. It could contain, for example, '30 (5.1.18-300.fc30.x86_64)'.

In offline mode the information is different as is is extracted from GRUB2 configurations (in an extremely straight-forward and error-prone way). The labelling of GRUB's menus could be very frivolous in the range of distributions, as well as location of grub.cfg file. Labels could even be extracted from grub.cfg into /boot/loader/entries (Fedora 30). Labels might not contain information about underlying OS version and architecture at all (Ubuntu 19.04).

The `architecture` field is a bit of a stretch in both modes because the architecture of the kernel is not necessarily the same as the hardware architecture (x86 kernel on x86_64 machine, for example). Which might or might not be the desired behaviour.

Also, all the information obtained with the help of grub.cfg is also pretty speculative because it reflects only the kernel that *might* be used by default.

Proposal: 

Use information about the OS from **/etc/os-release** with addition of utsname.sysname and utsname.release for `os_*` fields. Add version for all kernels present in the */boot/* directory (from respective **config-*** files) in offline mode.

Use utsname.machine in online mode and information from kernel **config-*** files in offline mode for `architecture` field.

The utsname.domainname and **/etc/hostname** are OK for `primary_host_name` field. No changes here.

The `interfaces` field is OK for online mode and does not make any sense for any of the offline modes. No changes here.

**Comment by jan-cerny** [comment]
@evgenyz This is a great analysis. I agree that the GRUB approach is error prone.

I'm fine with your proposal. It makes sense.

Fortunately, the [relevant section of OVAL specification](https://oval.mitre.org/language/version5.11/ovalresults/documentation/oval-system-characteristics-schema.html#SystemInfoType) gives us a lot of freedom regarding the contents of these elements.

**Comment by evgenyz** [comment]
Closed by #1505.
